sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2320,9 +2328,9 @@ case class Levenshtein(
 case class SoundEx(child: Expression)
   extends UnaryExpression with ExpectsInputTypes with NullIntolerant {
 
-  override def dataType: DataType = StringType
+  override def dataType: DataType = SQLConf.get.defaultStringType
```

Comment:
Again not sure if this should return default or pass through. Maybe some more input from others could help us resolve this discussion on when we want to do pass through and when we want to return default.

Reply:
I'm not sure about this one. `SoundEx` represents form of string encoding which I doubt should have to do anything with input type collation. For example, `Base64` drops input type collation.

Reply:
+1 for Nikola's reasoning

I would go with `SQLConf.get.defaultStringType` here, based on how this expression works, it seems to me that SoundEx result type doesn't really have anything to do with input type